### PR TITLE
Firestore - remove dead code

### LINKED ISSUE #136: Firestore - remove unused code from FirestoreTemplate and add unit tests for saveAll and delete
Continuation of #126

### PR DESCRIPTION
**Pull request by dmitry-s**
fixes #136

### CODE REVIEW

File: spring-cloud-gcp-data-firestore/src/test/java/com/google/cloud/spring/data/firestore/FirestoreTemplateTests.java [hunk 1]
```diff
@@ -150,6 +86,56 @@ public void findAllTest() {
...
+
+		CommitRequest.Builder builder = CommitRequest.newBuilder()
+				.setDatabase("projects/my-project/databases/(default)");
+
+		builder.addWrites(Write.newBuilder().setUpdate(buildDocument("e1", 100)).build());
+		builder.addWrites(Write.newBuilder().setUpdate(buildDocument("e2", 200)).build());
+
+
+		verify(this.firestoreStub, times(1)).commit(eq(builder.build()), any());
+		verify(this.firestoreStub, times(1)).commit(any(), any());
```
Comment (elefeint) [inline spring-cloud-gcp-data-firestore/src/test/java/com/google/cloud/spring/data/firestore/FirestoreTemplateTests.java:112]: Is the second verify redundant?

File: spring-cloud-gcp-data-firestore/src/test/java/com/google/cloud/spring/data/firestore/FirestoreTemplateTests.java [hunk 2]
```diff
@@ -150,6 +86,56 @@ public void findAllTest() {
...
+	}
+
+	@Test
+	public void deleteTest() {
+		doAnswer(invocation -> {
+			StreamObserver<CommitResponse> streamObserver = invocation.getArgument(1);
+			streamObserver.onNext(CommitResponse.newBuilder().build());
+			streamObserver.onCompleted();
+			return null;
+		}).when(this.firestoreStub).commit(any(), any());
```
Comment (elefeint) [inline spring-cloud-gcp-data-firestore/src/test/java/com/google/cloud/spring/data/firestore/FirestoreTemplateTests.java:122]: This wants to be a helper method.

**Review by elefeint — CHANGES_REQUESTED** [review]

**Comment by elefeint** [comment]
I've met this code. Code's, indeed, dead.

**Comment by elefeint** [comment]
Bigquery integration likely flake, config tests could go either way, but I am puzzled about kotlin tests. I never saw them fail before.

**Comment by ttomsu** [comment]
Config tests look to be a compile issue - does it work locally for you?

I'm seeing more of these dependency downloads failing. I've heard Maven central is pretty notorious for this kind of flakiness - JCenter allegedly has better reliability, but also is "less secure" because it allows authors to change the packages after publishing. I'm not sure why our GCS mirror isn't being chosen - I'll look into this.

**Review by meltsufin — APPROVED** [review]
LGTM modulo @elefeint's comments.

**Comment by dmitry-s** [comment]
@ttomsu yes it compiles locally.

**Review by elefeint — APPROVED** [review]

**Pull request merged**